File: solutions/2864. Maximum Odd Binary Number/2864.js
// Time complexity: O(n)
// Space complexity: O(1)
var maximumOddBinaryNumber = function(s) {
  const zeros = (s.match(/0/g) || []).length;
  const ones = s.length - zeros;
  return '1'.repeat(ones - 1) + '0'.repeat(zeros) + '1';
};

File: solutions/2864. Maximum Odd Binary Number/2864.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  string maximumOddBinaryNumber(string_view s) {
    const int zeros = ranges::count(s, '0');
    const int ones = s.length() - zeros;
    return string(ones - 1, '1') + string(zeros, '0') + '1';
  }
};